admin/src/js/services/select2-search.js

```diff
@@ -26,11 +27,31 @@ angular.module('inboxServices').factory('Select2Search',
       return $(format.sender(row.doc, $translate));
     };
 
-    const defaultTemplateSelection = function(row) {
-      if (row.doc) {
-        return row.doc.name + (row.doc.muted ? ' (' + $translate.instant('contact.muted') + ')': '');
+    const defaultTemplateSelection = function (selection) {
+      if (Array.isArray(selection)) {
+        return selection
+          .map(function (row) {
+            if (row.doc) {
+              return (
+                row.doc.name +
+                (row.doc.muted
+                  ? ' (' + $translate.instant('contact.muted') + ')'
+                  : '')
+              );
+            }
+            return row.text;
+          })
+          .join(', ');
       }
-      return row.text;
+      if (selection.doc) {
+        return (
+          selection.doc.name +
+          (selection.doc.muted
+            ? ' (' + $translate.instant('contact.muted') + ')'
+            : '')
+        );
+      }
+      return selection.text;
```

Comment:
This is extremely hard to follow. Was this a change requested by sonar? 
I think we can make this more readable: 
```suggestion
const defaultTemplateSelection = (selection) => {
  const formatRow = (row) => {
    if (!row.doc) {
      return row.text;
    }
    
    let formatted = row.doc.name;
    if (row.doc.muted) {
      formatted += `(${$translate.instant('contact.muted')})`;
    }
    return formatted;
  };
  if (Array.isArray(selection)) {
    return selection.map(row => formatRow(row)).join(', ');
  }
  return formatRow(selection);
}
```